zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1988,12 +1989,12 @@ public <T> T retryUntilConnected(final Callable<T> callable)
           retryCauseCode = e.code();
           // we give the event thread some time to update the status to 'Disconnected'
           Thread.yield();
-          waitForRetry();
+          waitForRetry(retryCount++);
```

Comment:
Just wondering, can we add count and Thread.sleep here?

Reply:
thanks for the review. there are multiple places where waitForRetry()is invoked and there are few other places where we exit right away. Encapsulating in waitForRetry() seems like a better choice.